### ISSUE TITLE
Completion percentage

### ISSUE THREAD

**Issue by cheesegrits** [transifex/txgh#18]
Not specifically a txgh issue, but (pursuant to our email exchange) something I wanted to get on the radar.

At the moment, the Transifex hook only fires once a translation is at 100%.  While I can see some logic to this, I think it is essential to make this a configurable setting in either Transifex or txgh, so we can specify the percentage.  This is after all a tool for continuous integration of Github and TX, so it seems a shame not to be able to actually use it to fully do that.

So either Transifex should fire the hook on every change and txgh decides what completion percentage it wants on a per-project basis, or the hook at Transifex should be configurable with the trigger percentage.

Thanks for listening.  :)

**Comment by matthewjackowski** [comment]
We'll need a polling mechanism in Txgh with added configuration that allows you to set how often to pull translations.  The 'percent complete' logic can then be added in a configurable way on the Txgh side without relying on Transifex.

**Comment by cheesegrits** [comment]
In another response, you said you didn't think modifying the webhook on the Transifex end to allow a configurable trigger percentage would get on the road map "any time soon".

Who would we need to lobby at Transifex to make that happen?

I'm not against polling in principle, but my project has over a thousand language files.  That's not really conducive to polling.

**Comment by matthewjackowski** [comment]
@cheesegrits You just lobbied.   We pay special attention to any requests coming from the community.    :+1:

I'll still keep the polling mechanism as an open issue because it allows the greatest flexibility in Txgh and can cover more generalized cases for smaller projects.  That allows the Transifex core web hooks to focus on more streamlined use cases.

**Comment by cheesegrits** [comment]
I totally agree that polling would be a very useful feature.  But for larger projects ... speaking of which, I actually I massively underestimated the number of language files we have.  I forgot we added a lot of new languages last year).  We now have a tad under 4000 ...

[vagrant@vagrant-centos65 joomla30]$ grep trans\. .tx/config | wc -l
3958

Having a working txgh has *massively* improved our Transifex integration, but our lives would be complete if we could get that option to trigger the Transifex hook on every change.

Thanks for all your work on this so far.

**Comment by cheesegrits** [comment]
Any word on this?

I'm doing another build of my product, going through my 3 monthly pain with pulling all 3156 translation files from Transifex with a manual client pull.

It'd be REALLY nice if we could get rid of that 100% thing ...

**Comment by matthewjackowski** [comment]
@cheesegrits 
Alright so we have had success getting this request on the roadmap to alter the webhook behavior in the core product.  But there is still some specification work todo before actual development can start.

What about going back to the polling idea?  That's something I can quickly whip into Txgh and while it won't be 'realtime'ish it should save you having todo manual pulls.

**Comment by ylavoie** [comment]
It would definetely help while we wait for realtime.

**Comment by ylavoie** [comment]
Any news on poll or realtime?
